pynxtools_xps/reader_utils.py

```diff
@@ -112,8 +112,13 @@ def to_snake_case(str_value):
 
 def convert_pascal_to_snake(str_value):
     """Convert pascal case text to snake case."""
-    pattern = re.compile(r"(?<!^)(?=[A-Z])")
-    return pattern.sub("_", str_value).lower()
+    # Convert CamelCase to snake_case
+    snake_case = re.sub(r"(?<!^)(?=[A-Z])", "_", str_value)
+
+    # Convert whitespace to underscores and remove extra underscores
+    snake_case_cleaned = re.sub(r"\s+", "_", snake_case).replace("__", "_")
```

Comment:
Shouldn't `\s+` not catch all whitespace anyways so that you can just do
```suggestion
    snake_case_cleaned = re.sub(r"\s+", "_", snake_case)
```
?

Reply:
I think it makes a difference because e.g. `Time per Spectrum Channel` would get converted to `time_per__spectrum__channel` with your solution and `time_per_spectrum_channel` with mine.

Reply:
```python
>>> import re
>>> snake_case = 'Time per Spectrum Channel'
>>> re.sub(r"\s+", '_', snake_case)
'Time_per_Spectrum_Channel'
>>> snake_case = 'Time  per  Spectrum Channel'
>>> re.sub(r"\s+", '_', snake_case)
'Time_per_Spectrum_Channel'
>>> snake_case = 'Time  per  Spectrum    Channel'
>>> re.sub(r"\s+", '_', snake_case)
'Time_per_Spectrum_Channel'
```

Is it different for you? 👀

Reply:
No, but these two are different:
```
import re

def fn_a(str_value):
    snake_case = re.sub(r"(?<!^)(?=[A-Z])", "_", str_value)
    # Convert whitespace to underscores and remove extra underscores
    snake_case_cleaned = re.sub(r"\s+", "_", snake_case).replace("__", "_")
    
    return snake_case_cleaned.lower()

def fn_b(str_value):
    snake_case = re.sub(r"(?<!^)(?=[A-Z])", "_", str_value)
    # Convert whitespace to underscores and remove extra underscores
    snake_case_cleaned = re.sub(r"\s+", "_", snake_case)
    
    return snake_case_cleaned.lower()

>>> fn_a("Time per Spectrum Channel")
time_per_spectrum_channel
>>> fn_b("Time per Spectrum Channel")
time_per__spectrum__channel
```

This is what we are discussing, right?

Reply:
Right. Sorry, I was too focused on this line and actually missed that the line above introduces underscores. Thanks for explaining :) I'll resolve